src/user-notification.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Factory for amp-user-notification
+ */
+
+import {getElementService} from './service';
+
+
+/**
+ * @param {!Window} window
+ * @return {!Promise<!UserNotification>}
+ */
+export function userNotificationFor(window) {
```

Comment:
It might be a good idea as well. However, here, I'm mainly reacting to the ambiguity: you are calling a single instance of the notification "AMP User  Notification" and you are calling a manager of multiple notifications "User Notification". That felt confusing to me. Alternative names could be "User Notification Service" or "User Notification Manager" or even simply plural - but the name should make it clear that it directs multiple notifications, I believe. There's no such issue with history or viewer or viewport - since they all designate a single history/viewer/viewport.